### ISSUE TITLE
Spanish Integration Bucket 2 - PWA

### ISSUE THREAD

**Issue by sonyekere** [episphere/connect#915]
Placeholder for requirements. @anthonypetersen will provide @bransteitterbr with development insight. Development to begin prior to receiving translated materials

March : Start Development 
-First, Build out Spanish version of MyConnect ( March to May) - Text translation/Hard text 
-Populate Spanish version of MyConnect with all content - (once content is received )
-Integrating Spanish versions of all forms (consent, HIPAA, revocation, data destruction)

**Comment by robertsamm** [comment]
Additional details below. Note, the majority of this work will be in the PWA but there are a few areas where this work will touch the SMDB (form versions - bubble fields and downloadable signed form, new CID to be added to SMDB)
1.  Build out Spanish Version of MyConnect and Hard Text Translation - refactor code to hold both in English and Spanish anywhere text is displayed on PWA and make sure it displays correct text for which language is selected. Should be able to toggle between English and Spanish. 
-translated text to be provided at later date. Actual build out can be started now with placeholders for the Spanish translation (e.g. could use Latin for placeholders). Est. date of translation and handover of content to be provided once timeline finalized.

2. Rendering of correct forms and surveys - survey rendering to be handled in issue  https://github.com/episphere/connect/issues/917. For rendering of correct forms, we will have separate site-specific Spanish translated consent forms and HIPAA forms versions for each site as well as a Spanish revocation form and destruction form (not site specific). If a participant selects 'Spanish' the version of the consent and HIPAA forms that should be displayed is the Spanish version in both the PWA and the SMDB. The Spanish versions will have separate variables. 
-PWA: render correct forms in PWA for all places forms are unsigned and signed
-SMDB: bubble field for forms versions, render correct form for downloadable forms on Participant Summary page
-still need decision: If someone signs consent form in Spanish should they always render in Spanish even if the person toggles back to English? 

3. CID for language election/preferred language - additional var for language election to be added that will be derived from which consent form the participant signs that indicates whether the individuals preferred language is English or Spanish. 
-SMDB: Add as bubble field on SMDB 'All Participants' page, Add on Participant Details Page as a variable that can be changed by NORC and CCC staff only
-PWA: we will not add this to the User Profile screen that the participant initially fills out. However, we would like to add this to the 'MyProfiles' page for a participant to be able to change. 
-API: allow sites to pull this information back via API
-Decision needed: should CSC and participant be able to change their language preference at any time or only after verification status=verified? Also confirm only CCC and NORC staff should be able to change language preference?

**Comment by depietrodeanna** [comment]
@bransteitterbr see below for Spanish HIPAA forms, data destruction, and HIPAA rev forms:

pSite HIPAA forms: https://nih.app.box.com/folder/256173575137
Data Destruction: https://app.box.com/s/xem0zgtqjqiftry3zfmu9qiu6jhhq7t0 
HIPPA Rev: https://app.box.com/s/i4eolku7f9sa3wsrdocv2hyl41jchtx6

**Comment by depietrodeanna** [comment]
json files for PWA here @bransteitterbr https://nih.app.box.com/file/1536067057824

**Comment by robertsamm** [comment]
Hi @bransteitterbr when you upload the forms into the PWA and SMDB, please use the below form version variable response labels for the Spanish forms. 

**Revocation & Destruction Forms (HIPAA Revocation Version CID: 407743866, Data Destruction Form Version CID: 304438543)**
HIPAA_Revocation_V1.0_Span
Data_Destruction_V1.0_Span

**Consent Forms (Consent Version CID: 454205108)**
HFHS_Consent_V0.03_Span
Marshfield_Consent_V0.03_Span
Sanford_Consent_V0.03_Span
UChicago_Consent_V0.05_Span
KPCO_Consent_V0.03_Span
KPNW_Consent_V0.03_Span
KPHI_Consent_V0.03_Span
KPGA_Consent_V0.04_Span
HP_Consent_V0.05_Span

**HIPAA Authorization Forms (HIPAA Version CID: 412000022)**
HFHS_HIPAA_V0.02_Span
Marshfield_HIPAA_V0.02_Span
Sanford_HIPAA_V0.02_Span
UChicago_HIPAA_V0.02_Span
KPCO_HIPAA_V0.02_Span
KPNW_HIPAA_V0.02_Span
KPHI_HIPAA_V0.02_Span
KPGA_HIPAA_V0.03_Span
HP_HIPAA_V0.03_Span

@bransteitterbr @anthonypetersen @brotzmanmj Madhuri and I discussed this one, and we wanted to know how do these vars get tied to the preferred language variable (CID 255077064 Consent and Notification Language). Is this done in the backend by DevOps? How does the preferred language var take the character responses from the form versions and transfer that to 0=English and 1=Spanish?

**Comment by depietrodeanna** [comment]
@bransteitterbr for the support page, we need the CSC specific line for Spanish to be displayed (we have both an email and a phone number).

Please use this contact info for the CSC contact info to render on the Spanish version of the PWA.
Phone: 1-877-775-7004
Email: ConnectAyuda@norc.org

**Comment by bransteitterbr** [comment]
@robertsamm Could the pdf versions of the Data Destruction and Revocation forms be added to the box folders where the docx sits? Thanks

**Comment by robertsamm** [comment]
@bransteitterbr PDF versions of forms uploaded and linked below (in same folders as Word versions):
Revocation: https://nih.app.box.com/file/1544011777350
Destruction: https://nih.app.box.com/file/1544017147039 

Model/NCI HIPAA: https://nih.app.box.com/file/1544015550916
pSite HIPAAs: https://nih.app.box.com/folder/256173575137

Model/NCI Consent: https://nih.app.box.com/file/1544022088783 (not yet IRB approved but can be uploaded for NCI site as will not move to stage/prod)

**Comment by robertsamm** [comment]
@bransteitterbr and @amber-emmes  for preferred language on the SMDB:
1. Please place the bubble field for preferred language on the 'All Participants' page between 'All Base Surv Complete' and 'Participation Status'
![image](https://github.com/episphere/connect/assets/67074735/0a35babe-b80e-4f12-a188-160cac8a314e)

2. Please place the editable preferred language field on the 'Participant Details' page after 'Connect ID' but before the section on phone/email login. Note, this is only editable for NORC and CCC staff. For site staff, the preferred language should display but the field would not be editable.
![image](https://github.com/episphere/connect/assets/67074735/ee06d45f-b8b9-42db-a5d8-eb1d8cf1dc57)

**Comment by depietrodeanna** [comment]
Starting to document fixes needed in PWA Spanish content, categorized into MVP and non-MVP (per my own categorization, others can comment if they disagree):

https://app.box.com/s/8kwt07qw46u6xyikl5a0v5uekbz81lza
@amber-emmes I believe you can help us since Brian is out but tagging @Davinkjohnson to confirm. 

Requested My Samples json updates from NORC for SF and UC today, they will deliver ASAP.

**Comment by robertsamm** [comment]
Good morning! I have also added some items to the Box doc linked above that I saw in my testing.

A few items that are MVP: 
1. Please change the PWA toggle label from 'Preferred Language' to just 'Language' to not confuse this with the preferred language variable that is not set by the toggle.
2. On SMDB - please change label on Pt Details page from 'Consent and Notification Language' to 'Preferred Language'. 
3. On SMDB - bubble field label on 'All Participants' page to be changed from 'Consent and Notification Language' to 'Preferred Language' with DD update to this CID. @mnataraj92 is this in the DD change log already?

**Comment by mnataraj92** [comment]
@rebexxxxxx is adding this to the DD change log!

**Comment by amber-emmes** [comment]
1) Has been addressed and merged into dev. It should be ready to re-test now.
2) Has been addressed and merged into dev. It should be ready to re-test now.
3) Will require some updates elsewhere and I believe will be handled through the DD change log.

**Comment by depietrodeanna** [comment]
continuing PWA testing. @amber-emmes for #1, the "Language" is showing up correctly in English but in Spanish I still see "Idioma preferido." Please update to just "Language" in Spanish: "Idioma"

**Comment by amber-emmes** [comment]
The Spanish-language update for 1 has been merged into dev and is ready for testing there.

**Comment by depietrodeanna** [comment]
> The Spanish-language update for 1 has been merged into dev and is ready for testing there.

tested and ready for stage

**Comment by depietrodeanna** [comment]
@sonyekere feel free to relocate if this is not the correct issue. NORC delivered the updated My Samples files for SF and UC: https://nih.app.box.com/folder/271691579589 
NORC said, "Please note that this folder also contains the individual en.js and es.js files along with an overview of the six places that Glyph joined sentences to translate. They have also updated the translation of the value proposition in the JSON file to match the glossary, as previously discussed."

@bransteitterbr let me know if you have everything you need here.

**Comment by cusackjm** [comment]
@bransteitterbr Spanish SSN Survey Content:
https://nih.app.box.com/file/1525593539781 (this is different than what is currently programmed)

**Comment by brotzmanmj** [comment]
@depietrodeanna the link that Julie just provided above for the IRB approved SSN survey content in Spanish. Is this what Brian should be using for programming or was another format provided by Glyph that was formatted for him to consume (with the same content)?

**Comment by depietrodeanna** [comment]
@brotzmanmj That should be the format for Brian. Is that the same format that you are using for programming other surveys @bransteitterbr? Glyph provided all translated PWA specific files here (https://nih.app.box.com/folder/262831674134) but those files don't include the actual survey instruments (they should include survey descriptions on the dashboard). The only survey Glyph partially translated was PROMIS.

**Comment by brotzmanmj** [comment]
Just to clarify, the SSN survey is the only one that is programmed directly in the PWA and therefore is part of the work Brian is doing. The rest of the surveys get programmed in Quest via IMS.

@bransteitterbr please let us know if the link Julie provided is what you need or if you have questions.

**Comment by bransteitterbr** [comment]
@depietrodeanna and @brotzmanmj I was able to update the language based on the document provided and will be submitting a PR with just that portion of the Spanish updates so that it can be tested in dev

**Comment by robertsamm** [comment]
@bransteitterbr and @amber-emmes The pSite consent forms are still with the IRB. They found some inconsistencies in the translations of the pSite consent forms where there were just a couple of words here and there that did not match the model consent form exactly. Can you confirm whether you already uploaded the HTML for each of the Spanish pSite consents into dev? If yes, we will go ahead and start working on getting together a list of all of the areas that will need to be changed in the HTML for each site's consent so that you have this. It's not much just a few words or phrases on each form.

**Comment by bransteitterbr** [comment]
@robertsamm Yes the Spanish pSite consent forms html is in dev.

**Comment by cusackjm** [comment]
Few spanish issues from Dev testing: https://github.com/episphere/connect/issues/1050 
Otherwise everything functioned as intended

**Comment by robertsamm** [comment]
@bransteitterbr @amber-emmes The model and pSite consent forms are all IRB approved. Links to the PDF version are below. I am working on compiling the changes that need to be made to the HTML and will add those in a separate comment.

PDF Consent Forms w/ IRB Stamps of Approval:

- Model (NCI) Spanish Consent Form: https://nih.app.box.com/file/1553767235511
- HP Spanish Consent Form: https://nih.app.box.com/file/1577111094468
- MF Spanish Consent Form: https://nih.app.box.com/file/1577130858151
- HFH Spanish Consent Form: https://nih.app.box.com/file/1577135327674
- SH Spanish Consent Form: https://nih.app.box.com/file/1577143175606
- UC Spanish Consent Form: https://nih.app.box.com/file/1577150793468
- BSWH Spanish Consent Form: https://nih.app.box.com/file/1577153811716
- KPCO Spanish Consent Form: https://nih.app.box.com/file/1577159068566
- KPNW Spanish Consent Form: https://nih.app.box.com/file/1577162943650
- KPGA Spanish Consent Form: https://nih.app.box.com/file/1577166241229
- KPHI Spanish Consent Form: https://nih.app.box.com/file/1577167213536

**Comment by robertsamm** [comment]
@bransteitterbr @amber-emmes Here are the changes that need to be made to all of the Spanish consent form HTMLs in PWA. Please let me know if you have any questions.
 
-Tracked versions of the changes needed are located here: https://nih.app.box.com/folder/273255021491 
-Not all sites need the same changes so here is a summary chart of the changes: 
![image](https://github.com/episphere/connect/assets/67074735/ba5a3e2c-bf89-499e-809c-752b6c9a7249)

**Comment by robertsamm** [comment]
> Hi @bransteitterbr when you upload the forms into the PWA and SMDB, please use the below form version variable response labels for the Spanish forms.
> 
> **Revocation & Destruction Forms (HIPAA Revocation Version CID: 407743866, Data Destruction Form Version CID: 304438543)** HIPAA_Revocation_V1.0_Span Data_Destruction_V1.0_Span
> 
> **Consent Forms (Consent Version CID: 454205108)** NCI_Consent_V0.05_Span HFHS_Consent_V0.03_Span Marshfield_Consent_V0.03_Span Sanford_Consent_V0.03_Span UChicago_Consent_V0.05_Span KPCO_Consent_V0.03_Span KPNW_Consent_V0.03_Span KPHI_Consent_V0.03_Span KPGA_Consent_V0.04_Span HP_Consent_V0.05_Span
> 
> **HIPAA Authorization Forms (HIPAA Version CID: 412000022)** NCI_HIPAA_V0.02_Span HFHS_HIPAA_V0.02_Span Marshfield_HIPAA_V0.02_Span Sanford_HIPAA_V0.02_Span UChicago_HIPAA_V0.02_Span KPCO_HIPAA_V0.02_Span KPNW_HIPAA_V0.02_Span KPHI_HIPAA_V0.02_Span KPGA_HIPAA_V0.03_Span HP_HIPAA_V0.03_Span
> 
> @bransteitterbr @anthonypetersen @brotzmanmj Madhuri and I discussed this one, and we wanted to know how do these vars get tied to the preferred language variable (CID 255077064 Consent and Notification Language). Is this done in the backend by DevOps? How does the preferred language var take the character responses from the form versions and transfer that to 0=English and 1=Spanish?
> 
> *Edited to add the NCI site version labels for the model consent and model HIPAA in dev.

These are all still correct, but was missing BSWH. Please use these labels for the BSWH consent and HIPAA form version variable responses:
-BSWH_Consent_V0.02_Span
-BSWH_HIPAA_V0.01_Span

**Comment by robertsamm** [comment]
@bransteitterbr just reviewed with Deanna and I missed one change for the HTML above. For KPCO please add a space between words 'usted.' and 'Brindar' on page 8. Added to the summary table of changes.
![image](https://github.com/episphere/connect/assets/67074735/7dc6284c-5b4c-4964-a314-4a28f25f9b99)

**Comment by robertsamm** [comment]
@bransteitterbr The BSWH Spanish consent form HTML still seems to be in English even when toggled to Spanish. The only Spanish text in the HTML seems to be the change for the CSC contact info. See screenshot below.

I haven't checked this for all sites yet, but the HTML for the BSWH HIPAA form is correctly showing as Spanish and both the consent + HIPAA HTMLs for the HP site is also correctly showing in Spanish so it doesn't seem to be impacting all site form HTMLs. As I continue through the rest of the sites in my testing I'll let you know if I find any addtl cases of this.

![image](https://github.com/user-attachments/assets/5365ecb8-53f7-4c9d-8561-d371ce3dc90a)

**Comment by robertsamm** [comment]
@bransteitterbr I found a couple inconsistencies in the HP Spanish HTML consent form that need to be corrected. See below. (HP HIPAA HTML looks good).

1.	‘Studio’ should be changed to ‘Estudio’ (pg 1)
 
![image](https://github.com/user-attachments/assets/205c6cd8-205b-41d2-a94d-0a25667e8424)

2.	‘in situ’ should be italicized to ‘in situ’ (section ¿POR QUÉ SE LLEVARÁ A CABO ESTE ESTUDIO?)
 
![image](https://github.com/user-attachments/assets/1d3fbbb5-f5e0-4a54-a8e4-88700548f52e)

3.	Spelling was cut off in ‘Internet’ and there should be a period and space added before ‘Cuando’. Should read ‘Complete encuestas por Internet. Cuando se sume…’ (section ¿QUÉ SUCEDERÁ DURANTE EL ESTUDIO?)
 
![image](https://github.com/user-attachments/assets/96d7e040-01c1-48a3-978e-ffb2ea8b6e7f)

4.	Remove extra period at the end of the sentence (section problemas y preguntas)
 
![image](https://github.com/user-attachments/assets/7bde6a9a-d9cf-4371-a5bc-0966fb828048)

5.	Add line of space between ‘Sirvase ingresar su nombre legal.’ And ‘Nombre’
 
![image](https://github.com/user-attachments/assets/f7f1a7af-1a9e-4477-975f-042b8ff33ac4)

**Comment by robertsamm** [comment]
@bransteitterbr very minor changes needed on Marshfield's HTML Consent form just to address some spacing issues. See below. (No issues found in the HIPAA HTML for MF)

1.	Please add line of space between last sentence ‘…detalles sobre Connect.’ And header ‘PROTECCIÓN DE LA CONFIDENCIALIDAD BRINDADA EN ESTE ESTUDIO’
 
![image](https://github.com/user-attachments/assets/02fd7e5d-86b1-4bbb-80e8-bbf2f99798c7)

2.	Please add line of space between ‘Sirvase ingresar su nombre legal.’ And ‘Nombre’ at the very end of doc
 
![image](https://github.com/user-attachments/assets/a2ebf2f0-553a-446d-b26c-a6d1c3fef8f8)

**Comment by robertsamm** [comment]
@bransteitterbr For Henry Ford Health, I found some inconsistencies in both the HTML consent and HTML HIPAA forms. See screenshots below.

**HFH HTML Consent**
1.	The telephone number for HFH is missing the digit ‘3’ at the beginning. It should be ‘313-874-6399’. 
a.	Is it also possible to put the last 4 digits of the phone number on the same line as the rest of the phone number? 
b.	And then can you put ‘Dirección de correo electrónico:’ on one line as well if it will fit?
![image](https://github.com/user-attachments/assets/8ebe59b7-1520-4947-bf8a-ec2683df5791)

2.	Please put ‘Algunas de esas afecciones…’ on same line as end of sentence with ‘…el riesgo de padecer cancer.’ As they are part of the same paragraph (section ¿POR QUÉ SE LLEVARÁ A CABO ESTE ESTUDIO?)
 
![image](https://github.com/user-attachments/assets/d8f58bc7-465a-45c8-8929-13126063eeaf)

3.	Please put ‘También podemos medir…’ on same line of text as ‘…como el colesterol.’ As they are part of the same paragraph. Should read as ‘...como el colesterol. También Podemos…’ medir’ (section ¿QUÉ HARÁN CON MI INFORMACIÓN Y MIS MUESTRAS? + ‘Muestras donadas.’
 
![image](https://github.com/user-attachments/assets/2c623fb6-ec8a-47c5-8739-50cde6338497)

4.	Please put ‘Sin embargo…’ on same line as ‘que sean …útiles en medicina.’ As they are in the same paragraph. (section ¿Hay alguna ventaja potencial para otras personas que pueda surgir del estudio?)
 
![image](https://github.com/user-attachments/assets/bf0e5ef3-e533-4e8a-881a-b03fc206c159)

5.	Please put ‘en Connect…’ on same line as ‘ …de aceptar participar’ as part of same sentence.  (section PAGO ¿Recibirá algún tipo de pago por participar en el estudio?)
 
![image](https://github.com/user-attachments/assets/9a8dd7c4-4b4d-43b3-824d-bb4393855914)

6.	Please add line of space between ‘Sirvase ingresar su nombre legal.’ And ‘Nombre’ at the very end of doc
![image](https://github.com/user-attachments/assets/9f812884-2b3e-4a85-b9b5-73547580b7cf)
 
**HFH HIPAA HTML**
1.	The second bullets ‘Información’ is missing an ‘I’ can you please add this in?
 
![image](https://github.com/user-attachments/assets/c93c15cc-9020-4bdb-adaa-d48f38c010da)

2.	Can you remove the ‘3’ at the end of the footer? It isn’t in the PDF HIPAA
 
![image](https://github.com/user-attachments/assets/17fe419b-4de3-42a9-b5b8-00fc600bd08b)

**Comment by robertsamm** [comment]
@bransteitterbr Please see feedback below for Sanford's consent form HTML. No issues found with the HIPAA HTML for SH. 

However, I did find an issue in the SH PDF consent form where there are a few sentences missing (looks correct in HTML) that we are going to have to get resubmitted to IRB. We plan to work with IRB to get this approval expedited so that we can hand off the revised PDF before stage testing.

SH Spanish consent form HTML issues:
1.	Please fix spacing between ‘…una pausa y’ and ‘regresar…’ as part of same sentence
 
![image](https://github.com/user-attachments/assets/ff43783f-59ca-41d7-9aad-133b6099580f)

2.	Please align the text in the paragraph ‘Nos permita invitarlo a donar más muestras o información que no se haya descrito ya.’ to the left. Right now it looks like a sub paragraph of the one below it but they are separate sections.
 
![image](https://github.com/user-attachments/assets/1aaf9c07-4b4e-4511-803e-e56b1e49cdb1)

3.	Please move ‘Los Registros Médicos Electrónicos’ to it’s own line underneath the large header ‘REGISTROS MÉDICOS ELECTRÓNICOS Y RESULTADOS DE LA INVESTIGACIÓN’

![image](https://github.com/user-attachments/assets/5fc2249f-8a64-4ea8-93f6-28db7a67409b)

4. Please add line of space between ‘Sirvase ingresar su nombre legal.’ And ‘Nombre’ at the very end of doc
![image](https://github.com/user-attachments/assets/691aed3f-5822-4558-b1bb-2c5f96e83f7f)

**Comment by bransteitterbr** [comment]
@robertsamm I have submitted a PR to address the issues with the BSWH, HFHS, HP, Marshfield, and Sanford documents

**Comment by kmazzilli** [comment]
@bransteitterbr we received IRB approval for the resubmitted Sanford Spanish consent form. The updated PDF is here (https://nih.app.box.com/file/1426280306262). The version variable response will remain the same and no other sites PDFs should need to be updated. Thanks!

**Comment by depietrodeanna** [comment]
@bransteitterbr noting Spanish PWA issues from Dev testing for review and adjustment: https://nih.app.box.com/file/1598358239116

**Comment by robertsamm** [comment]
@bransteitterbr Minor changes below needed for UChicago's consent form HTML.

1.	Please place sentence ‘Sin embargo, su participación es muy importante y podría contribuir…’ on same line of text as ‘…producir resultados que sean útiles en medicina.’ As are in same paragraph (section ¿Hay alguna ventaja potencial para otras personas que pueda surgir del estudio?)
 
![image](https://github.com/user-attachments/assets/a021ea51-2370-453c-be57-eafc73663d61)

2.	Please remove the line of space between the two bullet points ‘ Investigatores de su sistema de atención médica’ and ‘Colaboradores de investigación de otras instituciones…’ (section ‘PROTECCIÓN DE LA CONFIDENCIALIDAD BRINDADA EN ESTE ESTUDIO’)
 
![image](https://github.com/user-attachments/assets/db333ce8-703a-4d62-bd83-be21456c7747)

3.	Please add a line of space between ‘Sírvase ingresar su nombre legal.’ And ‘Nombre’  
![image](https://github.com/user-attachments/assets/9f526c94-62aa-4ae5-be63-c810e2c08e4c)

**Comment by robertsamm** [comment]
@bransteitterbr Feedback on KPCO consent HTML below. No changes needed for HIPAA HTML.

1.	It looks like there is an extra period after ‘ambientales).’ Could this be removed please? (activities section)
 
![image](https://github.com/user-attachments/assets/fe244b04-d557-4b73-9c55-7cf9a1e70d52)

2.	Please put these sentences on same line of text as part of same paragraph: ‘…su domicilio particular y otra información de contacto.’ and ‘Brindar su número de seguro social es opcional…’ (section ¿QUÉ SUCEDERÁ DURANTE EL ESTUDIO?)
 
![image](https://github.com/user-attachments/assets/a1e429e0-db71-473d-85f7-08d68e8a2b79)

3.	Please place these two sentences on same line of text as part of same paragraph: ‘…manera natural, como el colesterol.’ And ‘También podemos medir factores que afectan la salud…’ (section ¿QUÉ HARÁN CON MI INFORMACIÓN Y MIS MUESTRAS?)
 
![image](https://github.com/user-attachments/assets/54aa37bd-cb23-4208-8894-eb9a32e2cb42)

**Comment by robertsamm** [comment]
@bransteitterbr Feedback on KPNW HTML consent form below. No changes to HIPAA HTML needed. (first two issues below are also issues in KPCO form but 3rd is different)

1.	Please put these sentences on same line of text as part of same paragraph: ‘…su domicilio particular y otra información de contacto.’ and ‘Brindar su número de seguro social es opcional…’ (section ¿QUÉ SUCEDERÁ DURANTE EL ESTUDIO?)
 
![image](https://github.com/user-attachments/assets/50f59a17-3145-4bdd-a721-6e1c7b57ded3)

2.	Please place these two sentences on same line of text as part of same paragraph: ‘…manera natural, como el colesterol.’ And ‘También podemos medir factores que afectan la salud…’ (section ¿QUÉ HARÁN CON MI INFORMACIÓN Y MIS MUESTRAS?)
 
![image](https://github.com/user-attachments/assets/65bfc62c-5368-49c2-8d59-b6dda60bfd3f)

3.	It looks like there is an extra period at the end of this sentence. Can you please remove?: ‘El riesgo de que alguien lo identifique con dicha información es muy bajo.’ (section ¿Sus datos genómicos se compartirán con personas no pertenecientes a este estudio?)
 
![image](https://github.com/user-attachments/assets/abe5da69-1e8b-4bcf-bc04-facfdc418986)

**Comment by robertsamm** [comment]
@bransteitterbr feedback on the KPGA HTML forms below. There was one issue found in their consent and one issue found in the HIPAA.

KPGA consent HTML:
1.	There seems to be a mix of Spanish in English in this paragraph when toggled to Spanish, this should all be in Spanish. Can you look into this one? (section ¿POR QUÉ SE LLEVARÁ A CABO ESTE ESTUDIO?)
![image](https://github.com/user-attachments/assets/d8e43cd7-4ac0-4106-8375-1ccf693f03f9)

KPGA HIPAA HTML:
1.	This bullet point has some formatting issues, can you please fix? It should read all under one bullet point as such. Note the ‘no’ should also be underlined but : ‘Información de sus registros médicos, como diagnósticos, medicamentos, resultados de exámenes físicos, procedimientos y pruebas de laboratorio que se le hayan hecho antes y durante el estudio, incluidos resultados de pruebas genéticas, diagnósticos y tratamientos de salud mental y estado de VIH/sida. La información divulgada __no__ incluirá notas de psicoterapia ni diagnósticos o tratamientos de abuso de alcohol o drogas.’
![image](https://github.com/user-attachments/assets/fa64c87f-a851-44df-99a5-7acc05db6f53)

**Comment by depietrodeanna** [comment]
> @bransteitterbr noting Spanish PWA issues from Dev testing for review and adjustment: https://nih.app.box.com/file/1598358239116

Regarding the note about the value proposition on the landing page, after consulting with NORC team, I don't think the header was part of the programming shell file that the team translated. Can you confirm it's an image? I recall you mentioning this earlier but want to be sure I'm fully understanding what we have to do.

Also, it looks like Glyph did update the translation of the value proposition in the final [es.json](https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapp.box.com%2Ffile%2F1570203436285&data=05%7C02%7Cdeanna.depietro%40nih.gov%7C389e467a75ba49c664e408dcaa8192a1%7C14b77578977342d58507251ca2dc2b06%7C0%7C0%7C638572720724607206%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=z8PhaDpzhgrzs%2BpPWkbkvpZd2NGFwy9SFKKH9MnRze8%3D&reserved=0) file:

"home": {
        "titleText": "Conéctese <em>hoy.</em><br/>Prevenga el cáncer<br/><em>mañana.</em><br/><br/><img src=\"./images/newImages/ConnectLogo.png\" alt=\"Logotipo de Connect\"/><br/><br/>",

There are no remaining instances of “súmese” anywhere in that file. So that should be integrated into the final Spanish PWA.

**Comment by robertsamm** [comment]
@bransteitterbr Feedback on KPHI consent HTML below. No changes needed to the HIPAA HTML. Also, this is the last of my feedback for the site form HTMLs! Thanks Brian!

1.	Please fix spacking between sentences as part of same paragraph: ‘…Una meta importante de Connect es estudiar las condiciones que aumentan el riesgo de padecer cáncer.’ And ‘Algunas de esas afecciones son muy comunes y benigna…’ (section: ¿POR QUÉ SE LLEVARÁ A CABO ESTE ESTUDIO?)
 
![image](https://github.com/user-attachments/assets/584b03ad-33c0-4a00-9ae1-ce59c9afb573)

2.	The text between the sections ‘Restos no utilizados de muestras’ and ‘PAGO’ all seem to be indented one tab over under ‘Registros electrónicos de salud y otros registros médicos’ section. Can you please align the text to the left as they are not a subsection?
 
![image](https://github.com/user-attachments/assets/b45ca059-52d7-43a7-878a-bfe5f68525f7)

3.	Please remove the extra period after ‘medicina’ and then move the ‘Sin embargo, su articipación es muy importante…’ onto the same line of text after ‘medicina’ as part of the same paragraph.
![image](https://github.com/user-attachments/assets/4918753d-8049-4bc9-a677-b7ba11752f0d)

**Comment by cusackjm** [comment]
> Few spanish issues from Dev testing: #1050 Otherwise everything functioned as intended

SSN Survey passed dev testing (related issue is closed)

**Comment by robertsamm** [comment]
@bransteitterbr have the Spanish consent form PDFs been updated in the SMDB for NCI and all sites? It looks like so far these are showing as the English PDF placeholders still (from back when we were waiting on IRB approval of Span consents) even when consented in Spanish. I've only checked NCI and BSWH in the SMDB so far, still working through the other sites.

**Comment by robertsamm** [comment]
Good morning @bransteitterbr and @sonyekere I found an issue in the Spanish revocation and destruction forms where there was placeholder text left in signature section of the forms when translated which meant that when the form is signed the signature coordinates overlap that placeholder text. (see screenshot below). To fix this the Spanish revocation and destruction forms would need to be replaced in the PWA (1) unsigned PDF forms, (2) signed PDF forms, (3) unsigned HTML display, and in the SMDB (signed PDF forms). 

How easy of a fix is this? Do you think we could fit it in in this release or will it need to wait until the next release? Since we don't have any Spanish language participants yet we don't expect to have many, if any, Spanish revocation and destruction requests before the August release so it's not a must have.

![image](https://github.com/user-attachments/assets/4965cf96-710e-44d1-9292-a6a3a7256026)

I've created new copies of the revocation and destruction forms without the placeholder text here. The version variable responses would remain the same for each.
Spanish data destruction form: https://nih.app.box.com/file/1601769243874
Spanish revocation form: https://nih.app.box.com/file/1601773729887

**Comment by bransteitterbr** [comment]
@robertsamm This is a quick fix just get me a copy of the pdf without the placeholder text and I can drop it in to the PWA and SMDB quickly

**Comment by robertsamm** [comment]
@bransteitterbr Great! Here's the new copies without the placeholder text:
Spanish data destruction form: https://nih.app.box.com/file/1601769243874
Spanish revocation form: https://nih.app.box.com/file/1601773729887

**Comment by robertsamm** [comment]
@bransteitterbr The SH Spanish consent form has been updated to the updated form correctly, but the signature coordinates are now off. The coordinates are now landing about half a page up overlapping text. See screenshot below. Can this be fixed?

![image](https://github.com/user-attachments/assets/4d52be38-bec2-4081-9a02-e54e949d3e91)

**Comment by depietrodeanna** [comment]
@bransteitterbr the logo in the top right header of the PWA is available in Spanish here: file:///Users/depietroda/Library/Containers/com.microsoft.Outlook/Data/tmp/Outlook%20Temp/Connect_DCEG_Logo_white-SPANISH%20(2)%5B1%5D.svg

![Connect_DCEG_Logo_white-SPANISH (2)](https://github.com/user-attachments/assets/3981fdd2-74f3-4000-84da-a6ff554455da)

Please let me know if you have issues integrating.
<img width="520" alt="Screenshot 2024-07-25 at 8 59 25 AM" src="https://github.com/user-attachments/assets/553ace7c-714a-4ba3-91eb-fa9ae471845a">

**Comment by robertsamm** [comment]
@bransteitterbr I did some testing of the Spanish PWA on iPad device in Safari browser and found 2 issues here related to the e-consent headers and the 'Siguiente'/Next page buttons after consent but prior to UP. I've added them to Deanna's tracking doc here: https://nih.app.box.com/file/1598358239116

**Comment by robertsamm** [comment]
> @bransteitterbr The SH Spanish consent form has been updated to the updated form correctly, but the signature coordinates are now off. The coordinates are now landing about half a page up overlapping text. See screenshot below. Can this be fixed?
> 
> ![image](https://private-user-images.githubusercontent.com/67074735/352157064-4d52be38-bec2-4081-9a02-e54e949d3e91.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5MTg2NzksIm5iZiI6MTcyMTkxODM3OSwicGF0aCI6Ii82NzA3NDczNS8zNTIxNTcwNjQtNGQ1MmJlMzgtYmVjMi00MDgxLTlhMDItZTU0ZTk0OWQzZTkxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI1VDE0MzkzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA4Zjg3Y2NjMzkyZTBjYjQ1NDU0YmE1NmQ3MzY1Y2YwODUzNGUzMGFiYzA3YzYyZDlhZTc5MDc0NmFkMDI2ZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4ocoLHyi79bUp9Z8YIHvBBp5LFCCBrySntZexRBRMMw)

Same issue is also found in the SH Spanish consent form that is downloaded from the Pt Summary pg in SMDB

**Comment by robertsamm** [comment]
Update on forms testing in dev - all of these items look good in dev in both Spanish and English and are ready for stage: (1) all sites HTML consent and HIPAA forms in PWA, (2) all sites unsigned PDF consent and HIPAA forms in PWA, (3) all sites (except SH) signed PDF consent and HIPAA forms in PWA and SMDB, (4) revocation and destruction box display and unsigned PDF forms in PWA, (5) revocation and destruction signed PDFs on PWA and SMDB.

Only outstanding item related to forms testing is Sanford's signed consent form PDF where the coordinates are not in the correct place. This needs to be fixed and tested in both the PWA and SMDB. The HIPAA form looks good. I will be OOO on Friday, tagging @kmazzilli here to assist with testing Sanford's form once it is ready.

**Comment by bransteitterbr** [comment]
@robertsamm All of the pdf updates are merged to dev and ready to retest

**Comment by robertsamm** [comment]
@bransteitterbr Thanks Brian! I just finished testing and all looks good. Thanks for making those updates! Forms are all ready for stage.

**Comment by robertsamm** [comment]
@bransteitterbr I found one issue with the signature coordinates in stage for the HP consent form in the SMDB. See screenshot below. However, the PWA signature coordinates for HP looked good and I found no other issues with the signature coordinates for the other site forms. 

Given the participant facing signature coordinates in the PWA look good and the issue is only in the SMDB, if this fix has to wait until the next release I think that would be ok. @sonyekere 
![image](https://github.com/user-attachments/assets/717ef2b5-cca7-48e2-8fca-8da355ded0a4)

All other site forms - HTML, PDF, consent, HIPAA, revocation, and destruction all look good in stage. Preferred language variable looks good in stage. Tested on all browsers + computer, iPad, and iPhone and found no other issues.

**Comment by bransteitterbr** [comment]
@robertsamm and @sonyekere this was a typo on my part when I was moving over the coordinates from the PWA 492 instead of 392.  I have already started the PR for this so it shouldn't have to wait

**Comment by bransteitterbr** [comment]
@robertsamm The PR has been pushed to stage try testing it again after 3:40 to give the deployment time to finish

**Comment by robertsamm** [comment]
@bransteitterbr thanks for the quick fix! Just checked and it looks good. All set for prod for the forms testing!

**Comment by sonyekere** [comment]
@depietrodeanna Can you please confirm Prod status for the comms items?

**Comment by depietrodeanna** [comment]
All bucket PWA items that were MVP for launch are addressed and ready for prod.

**Comment by sonyekere** [comment]
In production
